src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/MetadataTest.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection.Metadata;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Emit;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.EditAndContinue.UnitTests
+{
+    internal partial class MetadataTest : IDisposable
```

Comment:
>MetadataTest [](http://example.com/codeflow?start=27&length=12)

Looks great, I like it! I'd name this type something more general through. We might want to validate IL and PDBs in additiona to metadata.

Perhpas `EditAndContinueTest` would do.